modbus/functions/mbfuncholding.c

```diff
@@ -182,7 +182,7 @@ eMBFuncReadHoldingRegister( UCHAR * pucFrame, USHORT * usLen )
         usRegAddress++;
 
         usRegCount = ( USHORT )( pucFrame[MB_PDU_FUNC_READ_REGCNT_OFF] << 8 );
-        usRegCount = ( USHORT )( pucFrame[MB_PDU_FUNC_READ_REGCNT_OFF + 1] );
+        usRegCount |= ( USHORT )( pucFrame[MB_PDU_FUNC_READ_REGCNT_OFF + 1] );
```

Comment:
Eh, it's a bug yes, if you expect to be able to use 16bits for register counts, but really, it's not actually correct to even _look_ at the first byte, not whether you include the first byte's values.  You are only allowed to have 0..125 for usRegCount, so _only_ the second byte is relevant.  .  You're only allowed to have 125 registers per request, per protocol, because even though there's 2 bytes for the this field, the _response_ only has a single byte for the number of _bytes_ in the response.  (the check below shows this)

Your change is _also_ correct, but it would be just as correct to simply drop the first line.

Reply:
Hello,

There are three other locations this applies to: lines 125, 253 and 260.

A range check immediately follows each pair of statements; ignoring the high byte would cause unexpected behavior when a too-large argument is provided.  ORing looks correct to me.

Or the high byte could be tested for zero separately, but this is equivalent, and costs clarity for not much improvement, if any, on various platforms (byte vs. word comparison, endianness (byte swapping)?).

Reply:
yes, but again, per specification, it's only allowed to be 0..125.  so you only need to bother looking at the lower byte.  It's "correct" to do the OR, I agree, but it's _also_ correct to just drop it :)
![mb-125-limit-chart](https://user-images.githubusercontent.com/127907/175789654-6a6d0572-be77-40de-9c0e-f7f9b41c9574.png)

and also the flowchat for validity checking:
![mb-125-limit-flowchart](https://user-images.githubusercontent.com/127907/175789641-aee055e0-c150-439a-afbe-ad11168e5f55.png)


Line 125 is handling the write multiple function, which per spec is only allowed to have the range 1..123, so could also just drop looking at the first byte.

line 253/260 handles read/write, which only allows 1..125 for the read and only 121 for the write, so again, could just drop looking at the first byte.

Yes, the "missing" or is _inconsistent_, but it's _not_ a bug.  Dropping the first byte checks entirely probably would have a minor code size improvement on 8bit platforms, virtually no difference anywhere else.